src/TuDa.CIMS.Web/Components/ShoppingCart/ShoppingSearch.razor.cs

```diff
@@ -31,14 +32,39 @@ private async Task AssetItemSelectedInternal(AssetItem item)
         await AssetItemSelected.InvokeAsync(item);
     }
 
+    /// <summary>
+    /// To filter for different types of Items
+    /// </summary>
+    private AssetItemType? _selectedAssetItemType;
+
     private async Task<IEnumerable<AssetItem>> Search(string nameOrCas, CancellationToken token)
     {
         if (string.IsNullOrWhiteSpace(nameOrCas))
         {
             return [];
         }
 
-        return await _assetItemApi.GetAllAsync(nameOrCas).Match(value => value, err => []);
+        var allItems = await _assetItemApi
+            .GetAllAsync(nameOrCas)
+            .Match(value => value, err => new List<AssetItem>());
+
+        switch (_selectedAssetItemType)
+        {
+            case AssetItemType.Chemical:
+                return allItems.Where(item => item is Chemical);
```

Comment:
ℹ️ This will not work like intended. Solvents are also chemicals, so you need to exclude them here.
```suggestion
            case AssetItemType.Chemical:
                return allItems.Where(item => item is Chemical and not Solvent);
```
(I don't know if this is valid c#)